### PR TITLE
Sorting cards by date

### PR DESCRIPTION
**Pull request by tahiya-mongo**
This PR sorts the media cards by their publish dates and places the most recently published contents on top.

### CODE REVIEW

File: src/components/dev-hub/card-list.js [hunk 1]
```diff
@@ -41,71 +41,76 @@ const getThumbnailUrl = media => {
...
-// TODO: Update this component to take one array of items and map to cards based
-// on the item type
+const sortCardsByDate = contentList =>
+    contentList.sort((a, b) => {
+        if (new Date(a.publishDate) > new Date(b.publishDate)) return -1;
+        if (new Date(a.publishDate) < new Date(b.publishDate)) return 1;
+        return 0;
+    });
+
+const showArticles = article => (
```
Comment (jestapinski) [inline src/components/dev-hub/card-list.js:51]: nit, but since these expect a single media object, can we rename to be singular? (same for videos, podcasts)
```suggestion
const showArticle = article => (
```

File: src/components/dev-hub/card-list.js [hunk 2]
```diff
@@ -41,71 +41,76 @@ const getThumbnailUrl = media => {
...
+    <ArticleCard
```
Comment (jestapinski) [inline src/components/dev-hub/card-list.js:52]: Do we want to add the `badge` props to these cards?

File: src/components/dev-hub/card-list.js [hunk 3]
```diff
@@ -41,71 +41,76 @@ const getThumbnailUrl = media => {
...
+        key={article['_id']}
+        image={withPrefix(article['atf-image'])}
+        tags={getTagLinksFromMeta(article)}
+        title={getNestedText(article['title'])}
+        description={getNestedText(article['meta-description'])}
+    />
+);
+
+const showVideos = video => (
+    <VideoModal
```
Comment (jestapinski) [inline src/components/dev-hub/card-list.js:63]: This can just be a `Card` with the `video` and `videoThumbnail` props now

File: src/components/dev-hub/card-list.js [hunk 4]
```diff
@@ -41,71 +41,76 @@ const getThumbnailUrl = media => {
...
+    <ArticleCard
+        key={podcast.title}
+        image={getThumbnailUrl(podcast)}
+        title={podcast.title}
+        description={podcast.description}
+    />
+);
+
+const showAllContent = item => {
+    if (item.mediaType)
```
Comment (jestapinski) [inline src/components/dev-hub/card-list.js:89]: This could be a nice candidate to use `switch`!

File: src/components/dev-hub/card-list.js [hunk 5]
```diff
@@ -41,71 +41,76 @@ const getThumbnailUrl = media => {
         : media.thumbnailUrl;
 };
 
-// TODO: Update this component to take one array of items and map to cards based
-// on the item type
+const sortCardsByDate = contentList =>
+    contentList.sort((a, b) => {
```
Comment (jestapinski) [inline src/components/dev-hub/card-list.js:45]:
```suggestion
    contentList.sort((a, b) => new Date(b.publishDate || b.pubdate) - new Date(a.publishDate || a.pubdate))
```

Since we are comparing with `-` we can shorten this to one line using a quick subtraction comparison between dates!

This also would not work for articles, since these support `pubdate` and not `publishDate`, so `||` chaining should do the trick.

File: src/components/dev-hub/card-list.js [hunk 6]
```diff
@@ -41,71 +41,76 @@ const getThumbnailUrl = media => {
...
+const showPodcasts = podcast => (
+    <ArticleCard
+        key={podcast.title}
+        image={getThumbnailUrl(podcast)}
+        title={podcast.title}
+        description={podcast.description}
+    />
+);
+
+const showAllContent = item => {
```
Comment (jestapinski) [inline src/components/dev-hub/card-list.js:88]: Can we rename this to something like `renderContentTypeCard`?

File: src/components/dev-hub/card-list.js [hunk 7]
```diff
@@ -41,71 +41,76 @@ const getThumbnailUrl = media => {
...
-                                        badge={video.mediaType}
-                                    />
-                                }
-                                thumbnail={getThumbnailUrl(video)}
-                            />
-                        ))}
-
-                {podcasts.length > 0 &&
-                    podcasts
+                {fullContentList.length > 0 &&
```
Comment (jestapinski) [inline src/components/dev-hub/card-list.js:110]: Do we need this check?

File: src/components/dev-hub/card-list.js (hunk 3 above)
Comment (tahiya-mongo) [reply src/components/dev-hub/card-list.js:63]: For some reason, changing the `image` prop to `videoThumbnail` doesn't render the thumbnail image. Weird because it seems to be working for the `index.js` page

File: src/components/dev-hub/card-list.js [hunk 8]
```diff
@@ -41,71 +40,79 @@ const getThumbnailUrl = media => {
         : media.thumbnailUrl;
 };
 
-// TODO: Update this component to take one array of items and map to cards based
-// on the item type
+const sortCardsByDate = contentList =>
+    contentList.sort((a, b) => {
+        return (
```
Comment (jestapinski) [inline src/components/dev-hub/card-list.js:45]: Even better since this is one line we can omit `{}` and `return`!

File: src/components/dev-hub/card-list.js [hunk 9]
```diff
@@ -41,71 +40,79 @@ const getThumbnailUrl = media => {
...
+        description={podcast.description}
+    />
+);
+
+const renderContentTypeCard = item => {
+    if (item.mediaType)
+        switch (item.mediaType) {
+            case 'youtube':
+            case 'twitch':
+                return showVideo(item);
```
Comment (jestapinski) [inline src/components/dev-hub/card-list.js:90]: Sorry to nitpick on these but can we use `render` instead of `show`?
```suggestion
                return renderVideo(item);
```

File: src/components/dev-hub/card-list.js [hunk 10]
```diff
@@ -41,71 +40,79 @@ const getThumbnailUrl = media => {
...
-                                key={podcast.title}
-                                image={getThumbnailUrl(podcast)}
-                                title={podcast.title}
-                                description={podcast.description}
-                                badge={podcast.mediaType}
-                            />
-                        ))}
+                {fullContentList
+                    .slice(0, visibleCards)
+                    .map(item => renderContentTypeCard(item))}
```
Comment (jestapinski) [inline src/components/dev-hub/card-list.js:115]:
```suggestion
                    .map(renderContentTypeCard)}
```

File: src/components/dev-hub/card-list.js (hunk 9 above)
Comment (tahiya-mongo) [reply src/components/dev-hub/card-list.js:90]: render definitely sounds better!

**Comment by tahiya-mongo** [comment]
Thanks Jordan!

**Comment by tahiya-mongo** [comment]
Thanks Jordan!
